include/zephyr/drivers/video-controls.h

```diff
@@ -1,43 +1,52 @@
-/**
- * @file
- *
- * @brief Public APIs for Video.
- */
-
 /*
  * Copyright (c) 2019 Linaro Limited.
+ * Copyright (c) 2024 tinyVision.ai Inc.
  *
  * SPDX-License-Identifier: Apache-2.0
  */
 #ifndef ZEPHYR_INCLUDE_VIDEO_CONTROLS_H_
 #define ZEPHYR_INCLUDE_VIDEO_CONTROLS_H_
 
+/**
+ * @file
+ *
+ * @brief Public APIs for Video.
+ */
+
 /**
  * @brief Video controls
  * @defgroup video_controls Video Controls
  * @ingroup io_interfaces
+ *
+ * The Video control IDs (CIDs) are introduced with the same name as
+ * Linux V4L2 subsystem and under the same class. This facilitates
+ * inter-operability and debugging devices end-to-end across Linux and
+ * Zephyr.
+ *
  * @{
  */
 
-#include <zephyr/device.h>
-#include <stddef.h>
-#include <zephyr/kernel.h>
-
-#include <zephyr/types.h>
-
 #ifdef __cplusplus
 extern "C" {
 #endif
 
 /**
  * @name Control classes
+ *
+ * List of control classes, to group related controls together in blocks.
+ *
+ * This list is kept identical to the Linux kernel definitions.
+ *
  * @{
  */
-#define VIDEO_CTRL_CLASS_GENERIC 0x00000000 /**< Generic class controls */
-#define VIDEO_CTRL_CLASS_CAMERA  0x00010000 /**< Camera class controls */
-#define VIDEO_CTRL_CLASS_MPEG    0x00020000 /**< MPEG-compression controls */
-#define VIDEO_CTRL_CLASS_JPEG    0x00030000 /**< JPEG-compression controls */
-#define VIDEO_CTRL_CLASS_VENDOR  0xFFFF0000 /**< Vendor-specific class controls */
+#define VIDEO_CID_BASE				0x00980900 /**< Base class controls */
+#define VIDEO_CID_CODEC_CLASS_BASE		0x00990900 /**< Stateful codec controls */
+#define VIDEO_CID_CAMERA_CLASS_BASE		0x009a0900 /**< Camera class controls */
+#define VIDEO_CID_FLASH_CLASS_BASE		0x009c0900 /**< Camera flash controls */
+#define VIDEO_CID_JPEG_CLASS_BASE		0x009d0900 /**< JPEG-compression controls */
+#define VIDEO_CID_IMAGE_SOURCE_CLASS_BASE	0x009e0900 /**< Image source controls */
+#define VIDEO_CID_IMAGE_PROC_CLASS_BASE		0x009f0900 /**< Image processing controls */
+#define VIDEO_CID_VENDOR_CLASS_BASE		0x00ff0900 /**< Vendor-specific class controls */
```

Comment:
I just rechecked this `VIDEO_CID_VENDOR_CLASS_BASE`. In fact, this is Zephyr-specific and does not exist in Linux.
Its equivalent counterpart in Linux should be [V4L2_CID_PRIVATE_BASE](https://elixir.bootlin.com/linux/v6.11.1/C/ident/V4L2_CID_PRIVATE_BASE). Could you change to this one (I also need it for the upcomming control framework PR) ? Thanks !

Reply:
Ah right, in `include/uapi/linux/videodev2.h` I did not check there!

```c
/*  User-class control IDs defined by V4L2 */
#define V4L2_CID_MAX_CTRLS              1024
/*  IDs reserved for driver specific controls */
#define V4L2_CID_PRIVATE_BASE           0x08000000
```

I pushed the change.